src/main/java/com/hltech/store/OracleEventStore.java

```diff
@@ -0,0 +1,145 @@
+package com.hltech.store;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+import javax.sql.DataSource;
+import java.io.ByteArrayInputStream;
+import java.nio.charset.StandardCharsets;
+import java.sql.Blob;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.function.Function;
+
+import static java.util.stream.Collectors.groupingBy;
+
+@Slf4j
+@RequiredArgsConstructor
+public class OracleEventStore<E> implements EventStore<E> {
```

Comment:
As I can see there only differences between oracle and postgre are save and extractEventsFromResultSet methods. Maybe we could create some kind of base class like SqlEventStore and put common things there (for both oracle and postgre).